### PR TITLE
Adds tests for functions in src/weighted_matrix.cpp

### PR DESCRIPTION
**Pull request by nikolase90**
Summarized
- Adds tests for functions in `src/weighted_matrix.cpp`, i.e. `tests/testthat/test-src_weighted_matrix.R`
- Adds documentation for functions in `src/weighted_matrix.cpp`. 
- Renames some of the variables inside `weight_matrix_cpp()`, so that they are in line with the paper. See section 3.2 in https://arxiv.org/pdf/1903.10464.pdf.

### CODE REVIEW

File: src/weighted_matrix.cpp [hunk 1]
```diff
@@ -13,44 +18,59 @@ using namespace Rcpp;
 // [[Rcpp::export]]
 arma::mat weight_matrix_cpp(List features, int m, int n, NumericVector w){
 
-    // Define variables
+    // Note that Z is a n x (m + 1) matrix, where m is the number
+    // of unique features. All elements in the first column are equal to 1.
+    // For j > 0, Z(i, j) = 1 if and only if feature j is present in
+    // the ith combination of features. In example, if Z(i, j) = 1 we know that
+    // j is present in l[i].
```
Comment (martinju) [inline src/weighted_matrix.cpp]: Should be "j is present in features[i]" ???

Comment (nikolase90) [reply src/weighted_matrix.cpp]: Yes!

Comment (nikolase90) [inline src/weighted_matrix.cpp]:
```suggestion
    // j is present in features[i].
```

Comment (nikolase90) [reply src/weighted_matrix.cpp]: Fixed with a5bea03.

**Pull request merged**